core/src/main/java/org/apache/spark/shuffle/api/ShuffleExecutorComponents.java

```diff
@@ -39,16 +39,20 @@
   /**
    * Called once per map task to create a writer that will be responsible for persisting all the
    * partitioned bytes written by that map task.
-   *  @param shuffleId Unique identifier for the shuffle the map task is a part of
+   * @param shuffleId Unique identifier for the shuffle the map task is a part of
+   * @param shuffleGenerationId The shuffle generation ID of the stage that this task belongs to,
```

Comment:
how about
```
The generation ID of the shuffle, which is always -1 if the shuffle map stage is deterministic. Otherwise, it starts with 0 and increases when the shuffle map stage gets retied.
```